src/CoreWCF.Primitives/src/CoreWCF/Telemetry/HttpRequestMessagePropertyWrapper.cs

```diff
@@ -0,0 +1,114 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Net;
+using System.Reflection;
+
+namespace CoreWCF.Telemetry;
+
+
+/// <summary>
+/// This is a reflection-based wrapper around the HttpRequestMessageProperty class. It is done this way so we don't need to
+/// have an explicit reference to System.ServiceModel.Http.dll. If the consuming application has a reference to
+/// System.ServiceModel.Http.dll then the HttpRequestMessageProperty class will be available (IsHttpFunctionalityEnabled == true).
+/// If the consuming application does not have a reference to System.ServiceModel.Http.dll then all http-related functionality
+/// will be disabled (IsHttpFunctionalityEnabled == false).
+/// </summary>
+internal static class HttpRequestMessagePropertyWrapper
+{
+    private static readonly ReflectedInfo? ReflectedValues = Initialize();
+
+    public static bool IsHttpFunctionalityEnabled => ReflectedValues != null;
+
+    public static string Name
+    {
+        get
+        {
+            AssertHttpEnabled();
+            return ReflectedValues!.Name;
+        }
+    }
+
+    public static object CreateNew()
+    {
+        AssertHttpEnabled();
+        return Activator.CreateInstance(ReflectedValues!.Type)!;
+    }
+
+    public static WebHeaderCollection GetHeaders(object httpRequestMessageProperty)
+    {
+        AssertHttpEnabled();
+        AssertIsFrameworkMessageProperty(httpRequestMessageProperty);
+        return ReflectedValues!.HeadersFetcher.Fetch(httpRequestMessageProperty);
+    }
+
+    private static ReflectedInfo? Initialize()
+    {
+        Type? type = null;
+        try
+        {
+            type = Type.GetType(
+                "System.ServiceModel.Channels.HttpRequestMessageProperty, System.ServiceModel, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089",
```

Comment:
These type names need to be updates to use CoreWCF namespaces and the correct assembly name